ihmc-communication/src/main/java/us/ihmc/communication/ros2/ROS2DemandGraphNode.java

```diff
@@ -4,20 +4,45 @@
 
 import java.util.ArrayList;
 import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.function.Consumer;
 
 public class ROS2DemandGraphNode
 {
    private final ArrayList<ROS2DemandGraphNode> dependents = new ArrayList<>();
    private final ROS2HeartbeatMonitor nodeHeartbeatMonitor;
+   private final Set<Consumer<Boolean>> demandChangedCallbacks = new HashSet<>();
+   private final AtomicBoolean wasDemanded = new AtomicBoolean(false);
 
    public ROS2DemandGraphNode(ROS2PublishSubscribeAPI ros2, ROS2Topic<Empty> heartbeatTopic)
    {
       nodeHeartbeatMonitor = new ROS2HeartbeatMonitor(ros2, heartbeatTopic);
+      nodeHeartbeatMonitor.setAlivenessChangedCallback(this::checkIfDemandChanged);
    }
 
-   public void addDependents(ROS2DemandGraphNode... dependents)
+   public void addDependents(ROS2DemandGraphNode... dependentsToAdd)
    {
-      this.dependents.addAll(Arrays.stream(dependents).toList());
+      dependents.addAll(Arrays.stream(dependentsToAdd).toList());
+      for (ROS2DemandGraphNode dependent : dependentsToAdd)
+         dependent.addDemandChangedCallback(this::checkIfDemandChanged);
+   }
+
+   public void addDemandChangedCallback(Consumer<Boolean> demandChangedCallback)
+   {
+      demandChangedCallbacks.add(demandChangedCallback);
+      for (ROS2DemandGraphNode dependent : dependents)
+         dependent.addDemandChangedCallback(this::checkIfDemandChanged);
+   }
+
+   private void checkIfDemandChanged(boolean ignored)
+   {
+      boolean isDemanded = isDemanded();
+      // if wasDemanded != isDemanded -> wasDemanded = isDemanded, trigger callbacks
+      if (wasDemanded.compareAndSet(!isDemanded, isDemanded))
```

Comment:
This line is confusing to me, so I added the above comment... but is the comment confusing too?

Reply:
Maybe write it out in plain english?